lib/containers/urls.pm

```diff
@@ -51,64 +51,220 @@ sub get_opensuse_registry_prefix {
+            totest         => sub { },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x']
+        },
+        '15-SP2' => {
+            released       => sub { 'registry.suse.com/suse/sle15:15.2' },
+            totest         => sub { },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x']
+        },
+        '15-SP3' => {
+            released       => sub { 'registry.suse.com/suse/sle15:15.3' },
+            totest         => sub { },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x']
+        }
+    },
+    microos => {
+        Tumbleweed => {
+            released => sub { 'registry.opensuse.org/opensuse/tumbleweed' },
+            totest   => sub {
+                'registry.opensuse.org/' . get_opensuse_registry_prefix . 'opensuse/tumbleweed';
+            },
+            available_arch => ['x86_64', 'aarch64', 'ppc64le', 's390x', 'arm']
+        },
+        '15.1' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.1' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.1/images/totest/containers/opensuse/leap:15.1';
+                } elsif (grep { $_ eq $arch } qw/aarch64 arm/) {
+                    'registry.opensuse.org/opensuse/leap/15.1/arm/images/totest/containers/opensuse/leap:15.1';
+                } else {
+                    '';
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        },
+        '15.2' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.2' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.2/images/totest/containers/opensuse/leap:15.2';
+                } elsif (grep { $_ eq $arch } qw/aarch64 arm/) {
+                    'registry.opensuse.org/opensuse/leap/15.2/arm/images/totest/containers/opensuse/leap:15.2';
+                } else {
+                    '';
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        },
+        '15.3' => {
+            released => sub { 'registry.opensuse.org/opensuse/leap:15.3' },
+            totest   => sub {
+                my $arch = shift;
+                if ($arch eq 'x86_64') {
+                    'registry.opensuse.org/opensuse/leap/15.3/images/totest/containers/opensuse/leap:15.3';
+                } elsif (grep { $_ eq $arch } qw/aarch64 arm/) {
+                    'registry.opensuse.org/opensuse/leap/15.3/arm/images/totest/containers/opensuse/leap:15.3';
+                } else {
+                    '';
+                }
+            },
+            available_arch => ['x86_64', 'aarch64', 'arm']
+        }
+    }
+);
+
+sub check_image_arch_in_registry {
```

Comment:
```suggestion
sub return_image {
```

Reply:
`check_image_arch_in_registry` is more descriptive. in the other side this function doesnt return any kind of image. it just checks whether the container image is built against the arch which is run on.

Reply:
OK, let's be descriptive: It does not check any registry, it checks hash. And the word `arch` does not fit there either.

What about `check_image_in_hash`?

Reply:
This sounds like a "tab or spaces" discussion :)
The function is basically checking if a given image exists in the hash which represents the registry contents... 

Honestly, not in favor of `check_image_in_hash` but I understand it's more descriptive... 

I can give you some examples that are descriptive and easy for the user to understand, but I would vote for simplicity and easiness. Why not `image_exists` (it returns true or false if agiven image exists or not).

Reply:
Neither to be honest.
`check_image_in_hash` => it is not that bad but it does reflect the purpose in my opinion. 
`image_exists` => we return a tuple of released/totest but there are cases that one of them can be empty
i will rename it `supports_image_arch` => the function is checking the `available_arch` which contains the arch that are supported for the image based on the distri and the version.

Reply:
*this change is ready for review*

Reply:
> `image_exists` => we return a tuple of released/totest but there are cases that one of them can be empty

I don't understand. I don't want you to return anything, just true or false as you are already doing :)

Reply:
> > `image_exists` => we return a tuple of released/totest but there are cases that one of them can be empty
> 
> I don't understand. I don't want you to return anything, just true or false as you are already doing :)

ignore. you can review the new name

Reply:
OK. cc: @jlausuch